src/debug.cc

```diff
@@ -534,7 +536,7 @@ CacheLogChannel::log(const DebugMessageHeader &header, const std::string &body)
         return;
 
     if (earlyMessages)
-        return (void)saveMessage(header, body);
+        return saveMessage(header, body);
 
     if (!TheLog.file())
         return;
```

Comment:
It worries me that, after recent changes, saveMessage() context started to vary so much:

* CacheLogChannel: The method is still called essentially unconditionally, regardless of whether the channel could log the message. The method remains the first level of defense against a premature logging. We could (after some minor code changes) for example, continue saving messages even after TheLog.file() starts returning a truthy value.
* The other two channels: The method is now called only for non-immediately-writeable messages.

I doubt cache_log is so special that it requires this unique approach. I suspect we ought to refactor this code to make it look similar to the other two streams, with comments to explain any differences among the three channels. If you agree, please refactor.

N.B. Clearly, CacheLogChannel::shouldWrite(void) is TheLog.file(). We just do not have that CacheLogChannel method (yet?).